src/cli.rs

```diff
@@ -197,19 +192,86 @@ fn should_eval(curr_cmd: &str) -> bool {
     all_sexps(SexpParser::new(None, curr_cmd)).is_ok()
 }
 
-fn run_command_in_scripting<W>(egraph: &mut EGraph, command: &str, mut output: W) -> io::Result<()>
+fn run_commands<W>(
+    egraph: &mut EGraph,
+    filename: Option<String>,
+    command: &str,
+    mut output: W,
+    mode: RunMode,
+) -> io::Result<Option<Error>>
 where
     W: Write,
 {
-    match egraph.parse_and_run_program(None, command) {
+    if mode == RunMode::ShowDesugaredEgglog {
+        return Ok(match egraph.resugar_program(filename, command) {
+            Ok(desugared) => {
+                for line in desugared {
+                    writeln!(output, "{line}")?;
+                }
+                None
+            }
+            Err(err) => {
+                log::error!("{err}");
+                Some(err)
+            }
+        });
+    };
+
+    Ok(match egraph.parse_and_run_program(filename, command) {
         Ok(msgs) => {
-            for msg in msgs {
-                writeln!(output, "{msg}")?;
+            if mode == RunMode::Interactive {
+                writeln!(output, "(done)")?;
             }
+            if mode != RunMode::NoMessages {
+                for msg in msgs {
+                    write!(output, "{msg}")?;
+                }
+            }
+            None
+        }
+        Err(err) => {
+            log::error!("{err}");
+            if mode == RunMode::Interactive {
+                writeln!(output, "(error)")?;
+            }
+            Some(err)
+        }
+    })
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Copy)]
+pub enum RunMode {
+    Normal,
+    ShowDesugaredEgglog,
+    Interactive,
+    NoMessages,
+}
+
+impl Display for RunMode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // A little bit unintuitive but RunMode is specified as command-line
+        // argument with flag `--mode`, so `--mode none` means a normal run.
```

Comment:
Do you mean `--mode normal`?

Reply:
Yes I'll fix this. Copied comment from old code and didn't update.